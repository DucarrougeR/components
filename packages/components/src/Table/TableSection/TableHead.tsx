import styled from 'styled-components'
import {
  tableSectionCSS,
  tableSectionDefaults,
  TableSectionProps,
} from './tableSection'

export const TableHead = styled.thead<TableSectionProps>`
  ${tableSectionCSS}
`

TableHead.defaultProps = tableSectionDefaults
