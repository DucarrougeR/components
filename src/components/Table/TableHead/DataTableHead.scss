.lens-data-table-head {
  text-align: left;
}
