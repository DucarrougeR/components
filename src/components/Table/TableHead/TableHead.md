### Table Headings

This component is used to define the heading of the table and describes the rows of data below. The `<TableHeader>` should have one `<TableRow>` tag and then use the `<TableHeaderCell>` tags to be semantically correct.

```js
<Table>
  <TableHead>
    <TableRow>
      <TableHeaderCell>Column 1</TableHeaderCell>
      <TableHeaderCell>Column 2</TableHeaderCell>
      <TableHeaderCell>Column 3</TableHeaderCell>
    </TableRow>
  </TableHead>
  <TableBody>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
    <TableRow>
      <TableDataCell>1</TableDataCell>
      <TableDataCell>2</TableDataCell>
      <TableDataCell>3</TableDataCell>
    </TableRow>
  </TableBody>
</Table>
```
