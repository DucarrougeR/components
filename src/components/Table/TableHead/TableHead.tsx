import styled from 'styled-components'

export const TableHead = styled.thead`
  text-align: left;
`
