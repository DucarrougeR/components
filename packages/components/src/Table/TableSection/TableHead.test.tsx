import 'jest-styled-components'
import React from 'react'
import { createWithTheme } from 'looker-components-test-utils'
import { TableHead } from './TableHead'

test('A <TableHead> should render', () => {
  const component = createWithTheme(<TableHead />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
