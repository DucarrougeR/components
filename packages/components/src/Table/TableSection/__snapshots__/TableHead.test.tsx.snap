// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <TableHead> should render 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  text-align: left;
}

<thead
  className="c0"
/>
`;
